vignettes/vignettes.Rmd

```diff
@@ -0,0 +1,159 @@
+# output:
+#   html_document:
+#     df_print: paged
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Vignette Title}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(#echo = TRUE,
+  collapse = TRUE,
+  comment = "#>")
+```
+
+## Introduction
+This vignette outlines a workflow of parsing and plotting structural variants from Variant Call Format (VCF) using the `StructuralVariantAnnotation` package. `StructuralVariantAnnotation` contains useful helper functions for reading and interpreting structural variants calls. The packages contains functions for parsing VCFs from a number of popular caller as well as functions for dealing with breakpoints involving two separate genomic loci encoded as `GRanges` objects. 
+
+## Installation
+
+The *StructuralVariationAnnotation* package can be loaded from *Bioconductor* as follows:
+
+```{r, eval=FALSE}
+if (!requireNamespace("BiocManager", quietly = TRUE))
+    install.packages("BiocManager")
+BiocManager::install("StructuralVariantAnnotation")
+```
+
+## Workflow
+### Data preparation and filtering
+VCF data is parsed into a `VCF` object using `readVCF` function from Bioconductor package `VariantAnnotation`. Details of `VCF` objects can be found by `browseVignettes("VariantAnnotation")`.
+```{r, warning=FALSE, message=FALSE}
+library(VariantAnnotation)
+vcf.file <- system.file("extdata", "gridss.vcf", package = "StructuralVariantAnnotation")
+vcf <- VariantAnnotation::readVcf(vcf.file, "hg19")
+vcf
+```
+
+Simple filters could be applied to a `VCF` object to remove unwanted calls.
+```{r, warning=FALSE,message=FALSE}
+# filter out low quality calls
+hq_vcf <- vcf[rowRanges(vcf)$FILTER %in% c(".", "PASS"),]
+# somatic calls have no support in the normal
+somatic_vcf <- vcf[geno(vcf)$QUAL[,"normal.bam"] == 0,]
+# somatic loss of heterozygosity has no support in the tumour
+loh_vcf <- vcf[geno(vcf)$QUAL[,"tumour.bam"] == 0,]
+```
+
+### Converting SV calls to GRanges
+The package employs `GRanges` objects for storage of breakends. Function `breakpointRanges()` converts SVs in a VCF to GRanges with one entry per breakend. A breakend on the '+' strand indicates a break immediately after the given position, and on the '-' strand indicates a break immediately before the given position. Unpaired variants are removed at this stage.
+
+```{r}
+#converting VCF object 'vcf' to GRanges object 'gr'
+suppressPackageStartupMessages(library(StructuralVariantAnnotation))
+gr <- breakpointRanges(vcf)
+gr
+```
+
+
+### Exploring breakpoints
+
+`findBreakpointOverlaps()` and `countBreakpointOverlaps()` are functions for finding and counting overlaps between GRanges objects, equivalent of `findOverlaps()` and `countOverlaps()`. All breakends must have their partner breakend included in the GRanges. A valid overlap requires that breakends on boths sides overlap.
+
+<!-- **EXAMPLE HERE FOR GENE OVERLAP** -->
```

Comment:
gene overlap would use `findOverlaps()`, not `findBreakpointOverlaps()`. `findBreakpointOverlaps()` is for comparing breakpoints to other breakpoint.